torch/_dynamo/variables/functions.py

```diff
@@ -185,7 +185,7 @@ def bind_args(self, parent, args, kwargs):
             itertools.count(), self.fn.__code__.co_freevars, closure
         ):
             if name == "__class__":
-                source = AttrSource(self.source, "__class__") if self.source else None
+                source = AttrSource(self.source.base, "__class__") if self.source else None
```

Comment:
No, I don't think I understand. self denotes a user function. I don't see why self.source is guaranteed to have a base; can't it be literally something like global source?

Reply:
No, not here. I'll add tests and then explain.